internal/tracing/tracers.go

```diff
@@ -0,0 +1,81 @@
+package tracing
+
+import (
+	"io"
+	"sync"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	_ "github.com/uber/jaeger-client-go"
+	jaegercfg "github.com/uber/jaeger-client-go/config"
+	"golang.org/x/xerrors"
+)
+
+type key int
+
+// ProtocolKey is the key used to denote a protocol in a `context.Context`.
+const ProtocolKey key = iota
+
+var (
+	// ProtocolTag specified the span tag used for denoting a protocol.
+	ProtocolTag = "protocol"
+	// UndefinedProtocol is the default ProtocolTag value used if no
+	// ProtocolKey is present in the context.
+	UndefinedProtocol = "__UNDEFINED_PROTOCOL__"
```

Comment:
wondering if just having "UNDEFINED" wouldn't do the job ?

Reply:
I reckon @pierluca's review remarked `undefined` to be ambiguous. I think `__UNDEFINED_PROTOCOL__` is fine - it stands out and might make it easier to notice because we'd want to name all protocols in the traces.